packages/schematics/angular/migrations/update-10/remove-solution-style-tsconfig.ts

```diff
@@ -35,11 +35,16 @@ export default function (): Rule {
   return (host, context) => {
     const logger = context.logger;
 
-    const files = new JSONFile(host, 'tsconfig.json').get(['files']);
-    if (!(Array.isArray(files) && files.length === 0)) {
-      logger.info('Migration has already been executed.');
+    try {
+      const files = new JSONFile(host, 'tsconfig.json').get(['files']);
 
-      return;
+      if (!(Array.isArray(files) && files.length === 0)) {
+        logger.info('Migration has already been executed.');
+
+        return;
+      }
+    } catch {
+      host.create('tsconfig.json', '');
```

Comment:
Maybe instead of create rename `tsconfig.base.json`?

```ts
    const tsConfigExists = host.exists('tsconfig.json');
    if (tsConfigExists) {
      const files = new JSONFile(host, 'tsconfig.json').get(['files']);
      if (!(Array.isArray(files) && files.length === 0)) {
        logger.info('Migration has already been executed.');

        return;
      }
    }

    if (host.exists('tsconfig.base.json')) {
      if (tsConfigExists) {
        host.overwrite('tsconfig.json', host.read('tsconfig.base.json') || '');
      } else {
        host.rename('tsconfig.base.json', 'tsconfig.json');
      }

      host.delete('tsconfig.base.json');
    }
```

Reply:
I changed it to your sugesstion but moved the `host.delete` line to be after the `host.overwrite` because it throws an error after a rename.